src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9022,6 +9038,17 @@ private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
                     opToken = SyntaxFactory.Token(opToken.GetLeadingTrivia(), kind, opToken2.GetTrailingTrivia());
                 }
 
+                // Parse non-associative operators as if they were right-associative (and put an error on the operator)
+                if ((newPrecedence == precedence) && IsNonAssociative(opKind))
+                {
+                    opToken = this.AddError(opToken, ErrorCode.ERR_UnexpectedToken, opToken.Text);
```

Comment:
Nit: might we be able to produce a more specific diagnostic/message?

Reply:
Do you have suggestions here? I'm not sure what the rmore-specific error could be.

Reply:
'..' is not associative?

Reply:
I'm a bit iffy about that... not sure we want to use that terminology in an error message, seems a little obscure to a beginner dev (or even just non-compiler folks).

I got curious about what other languages do here: turns out they just report this same error (`ERR_UnexpectedToken`) - well, at least the ones I tested.

```rust
error: expected one of `.`, `;`, `?`, or an operator, found `..`
 --> src/main.rs:2:17
  |
2 |     let x = 2..3..4;
  |                 ^^ expected one of `.`, `;`, `?`, or an operator here
```

```python
  File "<stdin>", line 1
    a[1:2:3:4] # needs 4 args, since a:b:c is step-by-c
           ^
SyntaxError: invalid syntax
```

```ruby
SyntaxError: (irb):1: syntax error, unexpected ..
1..2..3
      ^
```

If you feel strongly about using `Operator ".." is not associative` or something, poke me, but I'm not sure about that wording and I can't think of anything better

Reply:
Sounds fine. Thanks for checking other languages.